tests/aws/services/lambda_/test_lambda_integration_dynamodbstreams.py

```diff
@@ -411,67 +411,110 @@ def verify_failure_received():
-            (
-                {"id": {"S": "test123"}},
-                {"id": {"S": "test123"}, "id2": {"S": "42test"}},
+            pytest.param(
+                {"id": {"S": "id_value"}},
+                {"id": {"S": "id_value"}, "id2": {"S": "id2_new_value"}},
                 {"eventName": ["INSERT"], "eventSource": ["aws:dynamodb"]},
                 1,
+                id="content_multiple_filters",
             ),
-            # Test exists filter
-            (
-                {"id": {"S": "test123"}},
-                {"id": {"S": "test1234"}, "presentKey": {"S": "test123"}},
-                {"dynamodb": {"NewImage": {"presentKey": [{"exists": False}]}}},
+            # Test content filter using the DynamoDB data type "S"
+            pytest.param(
+                {"id": {"S": "id_value_1"}, "presentKey": {"S": "presentValue"}},
+                {"id": {"S": "id_value_2"}},
+                # Omitting the "S" does NOT match: {"dynamodb": {"NewImage": {"presentKey": ["presentValue"]}}}
+                {"dynamodb": {"NewImage": {"presentKey": {"S": ["presentValue"]}}}},
                 1,
+                id="content_filter_type",
             ),
-            # numeric filters
-            # NOTE: numeric filters seem not to work with DynamoDB as the values are represented as string
-            # and it looks like that there is no conversion happening
-            # I leave the test here in case this changes in future.
-            (
-                {"id": {"S": "test123"}, "numericFilter": {"N": "123"}},
-                {"id": {"S": "test1234"}, "numericFilter": {"N": "12"}},
-                {"dynamodb": {"NewImage": {"numericFilter": {"N": [{"numeric": [">", 100]}]}}}},
-                0,
-            ),
-            (
-                {"id": {"S": "test123"}, "numericFilter": {"N": "100"}},
-                {"id": {"S": "test1234"}, "numericFilter": {"N": "12"}},
-                {
-                    "dynamodb": {
-                        "NewImage": {"numericFilter": {"N": [{"numeric": [">=", 100, "<", 200]}]}}
-                    }
-                },
-                0,
+            # Test exists filter using the DynamoDB data type "S"
+            pytest.param(
+                {"id": {"S": "id_value_1"}, "presentKey": {"S": "presentValue"}},
+                {"id": {"S": "id_value_2"}},
+                # Omitting the "S" does NOT match: {"dynamodb": {"NewImage": {"presentKey": [{"exists": True}]}}}
+                {"dynamodb": {"NewImage": {"presentKey": {"S": [{"exists": True}]}}}},
+                1,
+                id="exists_filter_type",
             ),
+            # TODO: Fix native LocalStack implementation for exists
+            # pytest.param(
+            #     {"id": {"S": "id_value_1"}},
+            #     {"id": {"S": "id_value_2"}, "presentKey": {"S": "presentValue"}},
+            #     {"dynamodb": {"NewImage": {"presentKey": [{"exists": False}]}}},
+            #     2,
```

Comment:
This false negative took me forever to spot. A combination of bad test design and inappropriate configuration led to wrong assumptions!
It was configured to 1 but both events match against AWS. However, the second event never gets waited. Negative testing for the last event is unreliable!

Reply:
Will this test be reintroduced at a later point?